### DISCUSSION TITLE
Using mergerfs with Wine Prefix

### DISCUSSION THREAD

**Discussion by DocMAX** [trapexit/mergerfs#1384] [Q&A] [answered]
I try to sandbox changed files in a separate wine prefix like this:
`mergerfs -o category.create=ff "$game_dir/.wine":"$HOME/.wine"=RO "$PREFIX_FS" `
It kind of works, but at some time some files move from $HOME/.wine (which is read-only) to $game_dir/.wine.
System.reg, user.reg, userdef.reg move as soon as they get written. Why is that behavior and what options may i alter?
Thanks.

**Comment by trapexit** [comment]
I'm not sure what behavior you are expecting. If the software is rewriting the file atomically then naturally it will write to the only path you've allowed.

As explained in the docs...

https://github.com/trapexit/mergerfs?tab=readme-ov-file#how-it-works

> mergerfs does not support the copy-on-write (CoW) or whiteout behaviors found in aufs and overlayfs. You can not mount a read-only filesystem and write to it. However, mergerfs will ignore read-only filesystems when creating new files so you can mix read-write and read-only filesystems. It also does not split data across filesystems. It is not RAID0 / striping. It is simply a union of other filesystems.

**Comment by DocMAX** [comment]
the expected behavior is that $HOME/.wine will left untouched, and if any files get modified they go in $game_dir/.wine
much like overlayfs doesn't touch it's "lower dir". ok, maybe mergerfs is not supposed to work like overlayfs...

**Reply by trapexit** [reply, marked as answer]
> maybe mergerfs is not supposed to work like overlayfs...

Correct. As I describe in the docs mergerfs is *NOT* an overlay union filesystem. You should just use overlayfs.
